src/main/java/com/haxwell/apps/quizki/services/QuestionServiceImpl.java

```diff
@@ -38,11 +42,13 @@
 	private Set<String> topicStrings = new HashSet<String>();
 	
 	private CreatedQuestionDTO outputDTO;
+	private ArrayList<CreatedQuestionDTO> outputDTOs;
 	private Set<CreateChoiceDTO> choiceDTOs;
 	
 	private Question question;
 	private Question savedQuestion;
 	private Optional<Question> outQuestion;
+	private Page<Question> questions;
```

Comment:
Could `questions` be a local variable? Generally, a service is a collection of functions, business logic related operations, but the service, does not keep state. Whatever is using the service, keeps the state. Generally. Anyway, so if this was not on the class level, but instead it was on the method level, it wouldn't be state, and this function would still work. yeah?

Reply:
Just a habit I have from the embedded days where most vars were global (no classes either) local is better.

Reply:
Yes, now local.